### ISSUE TITLE
`excessive_precision` adds underscores that weren't there before

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust-clippy#6232]
I tried this code:

```rust
pub const Y: f64 = 0.33333333333333334;
```

I expected to see this happen: `excessive_precision` replaces it with `0.3333333333333333`.

Instead, this happened: `excessive_precision replaces it with `0.333_333_333_333_333_3`.

### Meta

- `cargo clippy -V`: clippy 0.0.212 (ffa2e7a 2020-10-24)
- `rustc -Vv`:
```
rustc 1.49.0-nightly (ffa2e7ae8 2020-10-24)
binary: rustc
commit-hash: ffa2e7ae8fbf9badc035740db949b9dae271c29f
commit-date: 2020-10-24
host: x86_64-unknown-linux-gnu
release: 1.49.0-nightly
LLVM version: 11.0
```

**Comment by mikerite** [comment]
This is intentional. Otherwise, the suggestion would trigger [unreadable_literal](https://rust-lang.github.io/rust-clippy/master/index.html#unreadable_literal). Generally the suggestions try not to trigger other lints.

**Comment by jyn514** [comment]
@mikerite but that lint was already triggering before, and in fact I had `-A clippy::unreadable_literal` on at the time. I would prefer to be consistent with the existing code rather than have cascading changes from a single fix.

**Comment by flip1995** [comment]
Usually out policy is to suggest the best possible code (in Clippy's opinion), to prevent triggering another lint on fixed code. But it is  also ok if fixed code triggers a new lint, if the suggestion generation would otherwise be too hard.

I'm not sure, if we should change the suggestion behavior of this lint. We could hack, that it only suggests adding underscores, if the `unreadable_literal` lint is enabled.

**Comment by camsteffen** [comment]
> cascading changes from a single fix

I agree with jyn514. It's simpler for both the user and for Clippy if each lint just stays in its own lane.
